js/ai/src/generate/action.ts

```diff
@@ -267,9 +275,22 @@ async function generate(
 async function actionToGenerateRequest(
   options: z.infer<typeof GenerateUtilParamSchema>,
   resolvedTools?: ToolAction[],
-  resolvedFormat?: Formatter
+  resolvedFormat?: Formatter,
+  modelInfo?: ModelInfo
 ): Promise<GenerateRequest> {
-  const out = {
+  if ((options.tools?.length ?? 0) > 0 && !modelInfo?.supports?.tools) {
+    logger.warn(
+      `The model does not support tools (you set: ${options.tools?.length} tools). ` +
+        'The model may not behave the way you expect.'
+    );
+  }
+  if (options.toolChoice && !modelInfo?.supports?.toolChoice) {
+    logger.warn(
+      `The model does not support toolChoice option (you set: ${options.toolChoice}). ` +
+        'The model may not behave the way you expect.'
```

Comment:
Log the model name here, also it should only log the warning if `modelInfo.supports` exists.

Reply:
Done.